SparkPost/matchbox
# FE-771 Restyle Button Component

### What Changed
- Restyles the Button and Button.Group components with styled-components
- Matches [these Invision mockups ](https://sparkpost.invisionapp.com/share/FJUVWXX7BXP#/393107481_Button_Symbols)
- Makes use of the deprecate prop helper on:
  - `plain`
  - `Component`
  - `primary`
- Replaces `component` proptype with `elementType` to support strings such as `div` or `a`
- Removes the automatic blur of buttons after clicking, to better support native functionality & accessibility
- Makes updates to Panel section actions, Page primary actions, and Empty State primary actions
- Does NOT add any styled-system props to Button

![Screen Shot 2020-01-07 at 10 56 08 AM](https://user-images.githubusercontent.com/3903325/71908609-59a4e900-313c-11ea-9e9d-8b15c5d1d69e.png)

![Screen Shot 2020-01-07 at 10 59 48 AM](https://user-images.githubusercontent.com/3903325/71908898-d932b800-313c-11ea-8d4d-9e5d4b0f0cbc.png)

### How To Test or Verify
- Run Storybook with `npm run start:storybook`
- Visit the following stories to verify:
  - Button
  - Page (primary action)
  - Empty State (primary action)
  - Panel (section actions)

### Note
- There is one button style missing from this PR, to be used as connected components to TextFields

### PR Checklist
